consensus/misc/eip4844.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2023 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package misc
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/params"
+)
+
+// CalcBlobFee calculates the blobfee from the header's excess data gas field.
+func CalcBlobFee(excessDataGas *big.Int) *big.Int {
+	// If this block does not yet have EIP-4844 enabled, return the starting fee
+	if excessDataGas == nil {
+		return big.NewInt(params.BlobTxMinDataGasprice)
+	}
+	return fakeExponential(big.NewInt(params.BlobTxMinDataGasprice), excessDataGas, big.NewInt(params.BlobTxDataGaspriceUpdateFraction))
```

Comment:
```suggestion
	return fakeExponential(params.BlobTxMinDataGasprice, excessDataGas, params.BlobTxDataGaspriceUpdateFraction)
```
Let's not be so defensive...

Reply:
Ah, hmm, unsure why this was like that.

Reply:
Ha no, those are ints, not big ints. It's not copy, it's costructor

Reply:
Ah my bad